com.unity.visualeffectgraph/Editor/GraphView/Views/Properties/ObjectPropertyRM.cs

```diff
@@ -64,9 +65,17 @@ public override void UpdateGUI(bool force)
             try
             {
                 m_Value = (UnityObject)obj;
-                m_ValueIcon.image = obj != null
-                    ? AssetPreview.GetMiniTypeThumbnail(m_Value)
-                    : AssetPreview.GetMiniTypeThumbnail(m_Provider.portType);
+                if (m_Provider.portType == typeof(ShaderGraphVfxAsset))
+                {
+                    //m_ValueIcon.image = AssetPreview.GetMiniTypeThumbnail(typeof(Shader));
+                    m_ValueIcon.image = Resources.Load<Texture2D>("Icons/sg_graph_icon");
+                }
```

Comment:
This is a hack to display the proper object icon for shader graph. But ideally it should work the same way as other types of objects.
I asked [here](https://unity.slack.com/archives/C47JPNNTZ/p1643731047040519) on slack for help, but I got no answer, so I decided to workaround.